### PR TITLE
Create new structure for Indiana dependent exemptions

### LINKED ISSUE #5223: Create new structure for Indiana dependent exemptions
For separate project, create new dependent exemption parameter to test separate effect

### PR DESCRIPTION
**Pull request by samantharudra**
Fixes #5223

### CODE REVIEW

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+- bump: minor
+  changes:
+    changed: 
+    - Modified in_base_exemptions.py to separate out dependents and spouse/head.
+    - Modified tests in in_base_exemptions.yaml.
+    added:
+    - Added more tests.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:7]:
```suggestion
    - Separate Indiana dependent exemptions.
```

File: policyengine_us/parameters/gov/states/in/tax/income/exemptions/base/amount.yaml [hunk 2]
```diff
@@ -9,6 +9,7 @@ metadata:
   reference:
     - title: IC 6-3-1-3.5 "Adjusted gross income" (a)(3)-(4)
       href: http://iga.in.gov/legislative/laws/2021/ic/titles/006#6-3-1-3.5
+    - title: 2023 IT-40 Income Tax Form Instruction Booklet, pg. 30
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/exemptions/base/amount.yaml:12]: also an href (hyperlink)

Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/exemptions/base/amount.yaml:12]:
```suggestion
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/in/tax/income/exemptions/base/amount.yaml:12]: Dont need the same reference multiple times

File: policyengine_us/parameters/gov/states/in/tax/income/exemptions/base/dependent.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+description: Indiana provides this base exemption amount for dependent.
+values:
+  2021-01-01: 1_000
+
+metadata:
+  period: year
+  unit: currency-USD
+  label: Indiana exemptions base amount - dependent
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/exemptions/base/dependent.yaml:8]:
```suggestion
  label: Indiana dependent exemptions amount 
```

File: policyengine_us/parameters/gov/states/in/tax/income/exemptions/base/dependent.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  2021-01-01: 1_000
+
+metadata:
+  period: year
+  unit: currency-USD
+  label: Indiana exemptions base amount - dependent 
+  reference:
+    - title: IC 6-3-1-3.5 "Adjusted gross income" (a)(3)-(4)
+      href: http://iga.in.gov/legislative/laws/2021/ic/titles/006#6-3-1-3.5
+    - title: 2023 IT-40 Income Tax Form Instruction Booklet, pg. 30
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/exemptions/base/dependent.yaml:12]:
```suggestion
```

File: policyengine_us/parameters/gov/states/in/tax/income/exemptions/base/dependent.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+values:
+  2021-01-01: 1_000
+
+metadata:
+  period: year
+  unit: currency-USD
+  label: Indiana exemptions base amount - dependent 
+  reference:
+    - title: IC 6-3-1-3.5 "Adjusted gross income" (a)(3)-(4)
+      href: http://iga.in.gov/legislative/laws/2021/ic/titles/006#6-3-1-3.5
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/exemptions/base/dependent.yaml:11]: could you double check that all references show the number

File: policyengine_us/parameters/gov/states/in/tax/income/exemptions/base/dependent.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
+description: Indiana provides this base exemption amount for dependent.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/exemptions/base/dependent.yaml:1]:
```suggestion
description: Indiana provides this base exemption amount for each dependent.
```

File: policyengine_us/tests/policy/baseline/gov/states/in/tax/income/exemptions/in_base_exemptions.yaml [hunk 7]
```diff
@@ -1,15 +1,35 @@
...
     in_base_exemptions: 3_000
+
+- name: Married filing joint, no dependents 
+  period: 2021
+  input:
+    filing_status: JOINT
+    tax_unit_dependents: 0
+    state_code: IN
+  output:
+    ga_exemptions: 2_000
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/in/tax/income/exemptions/in_base_exemptions.yaml:26]: need to test in_base_exemptions

File: policyengine_us/tests/policy/baseline/gov/states/in/tax/income/exemptions/in_base_exemptions.yaml [hunk 8]
```diff
@@ -1,15 +1,35 @@
...
     state_code: IN
-    tax_unit_size: 3
   output:
     in_base_exemptions: 3_000
+
+- name: Married filing joint, no dependents 
+  period: 2021
+  input:
+    filing_status: JOINT
+    tax_unit_dependents: 0
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/in/tax/income/exemptions/in_base_exemptions.yaml:23]: could you create also some tests with available dependents

File: policyengine_us/variables/gov/states/in/tax/income/exemptions/in_base_exemptions.py [hunk 9]
```diff
@@ -10,6 +10,22 @@ class in_base_exemptions(Variable):
...
+
+	
+	# Dependent exemptions
+        dependents = tax_unit(
+            "tax_unit_dependents", period
+        )  # Total the number of dependents
+
+        dependent_exemptions = dependents * p.base.dependent
+
+
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/in/tax/income/exemptions/in_base_exemptions.py:29]: delete white space

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Comment by PavelMakarchuk** [comment]
To run the test:
```
policyengine-core test ./policyengine_us/tests/policy/baseline/gov/states/in/tax/income/exemptions/in_base_exemptions.yaml
```

File: changelog_entry.yaml [hunk 10]
```diff
@@ -0,0 +1,6 @@
+- bump: minor
+  changes:
+    changed: 
+    - Separate Indiana dependent exemptions. 
+    
+
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:6]:
```suggestion

```

Comment (PavelMakarchuk) [reply changelog_entry.yaml:6]: We should only have one empty line which starts all the way on the left

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/in/tax/income/exemptions/in_base_exemptions.py [hunk 11]
```diff
@@ -12,4 +12,14 @@ class in_base_exemptions(Variable):
...
     def formula(tax_unit, period, parameters):
         p = parameters(period).gov.states["in"].tax.income.exemptions
-        return tax_unit("tax_unit_size", period) * p.base.amount
+        # Personal exemptions
+        non_dependent_exemptions = p.base.amount
+        # Dependent exemptions
+        dependents = tax_unit("tax_unit_dependents", period)
+        unit_size = tax_unit("tax_unit_size", period)
+        unit_without_dependents = max_(unit_size - dependents, 0)
+        total_base = unit_without_dependents * non_dependent_exemptions
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/in/tax/income/exemptions/in_base_exemptions.py:21]: Please rename `base/amount` to `base/filer`. Also please avoid adding new Python variables that only rename parameters, as the absence of `p.` makes it less obvious that it's a parameter.
```suggestion
        total_base = unit_without_dependents * p.base.filer
```

File: policyengine_us/variables/gov/states/in/tax/income/exemptions/in_base_exemptions.py [hunk 12]
```diff
@@ -12,4 +12,14 @@ class in_base_exemptions(Variable):
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/in/tax/income/exemptions/in_base_exemptions.py:12]: Please split the formula into these components for clarity. In general we try to apply this in cases where the formula ends in a summation.
```suggestion
    adds = ["in_base_filer_exemptions", "in_base_dependents_exemptions"]
```

File: policyengine_us/parameters/gov/states/in/tax/income/exemptions/base/dependent.yaml [hunk 13]
```diff
@@ -0,0 +1,15 @@
+description: Indiana provides this base exemption amount for each dependent.  
+values:
+  2021-01-01: 1_000
+
+metadata:
+  period: year
+  unit: currency-USD
+  label: Indiana dependent exemptions amount 
+  reference:
+    - title: IC 6-3-1-3.5 "Adjusted gross income" (a)(3)-(4)
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/in/tax/income/exemptions/base/dependent.yaml:10]: (3) is the filer amount not dependent.
```suggestion
    - title: IC 6-3-1-3.5 "Adjusted gross income" (a)(4)
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Comment by PavelMakarchuk** [comment]
Not needed due to #5353

**Pull request closed without merging**